### PR TITLE
Transfer learning

### PR DESCRIPTION
**Pull request by SinaSarparast**
- Adds a pytorch lightning data loader module for the Omniglot dataset.

### CODE REVIEW

File: image_classification_simulation/data/omniglot_loader.py [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+            ]
+        )
+
+        self.test_set_transformation = transforms.Compose(
+            [
+                # Omniglot images have 1 channel,
+                #  but our model will expect 3-channel images
+                transforms.Grayscale(num_output_channels=3),
+                transforms.Resize(
+                    [int(self.image_size * 1.15), int(self.image_size * 1.15)]
```
Comment (alzaia) [inline image_classification_simulation/data/omniglot_loader.py:65]: Why are we resizing specifically with a 1.15 factor?

**Review by alzaia — APPROVED** [review]
Looks good.

File: image_classification_simulation/data/omniglot_loader.py (hunk 1 above)
Comment (SinaSarparast) [reply image_classification_simulation/data/omniglot_loader.py:65]: The resizing is a part of the data loader used in one of the tutorials. I think it is some sort of zooming in and upsampling. I don't how it affects the evaluation since it is done only to the test data. I'm going to set everything up in the repo and start experimenting with different things such as removing all of the augmentations to see how things change.

Comment (alzaia) [reply image_classification_simulation/data/omniglot_loader.py:65]: Yeah a lot of these tutorials on omniglot seem to use the same processing (also going from 1 to 3 channels even if it's grayscale) to adapt to the architecture used. For now it's fine, we will explore these transforms more later.

**Comment by SinaSarparast** [comment]
In This PR:

- Flower102, Office31, Omniglot data loaders are added
- Doc strings have been modified to conform to the Numpy convention.
- The option to choose data, and learning rate are added as hyper-parameters. 
- Minor updates are made to the Flake8 and black config files.

**Review by alzaia — APPROVED** [review]
